File: libs/jgdsl/src/gdsl/asm/operand/Memory.java
package gdsl.asm.operand;

import gdsl.asm.Visitor;

public class Memory extends Operand {
  private Operand pointer;
  
  public Operand getPointer() {
    return pointer;
  }

  /**
   * @param pointer
   */
  public Memory (Operand pointer) {
    super();
    this.pointer = pointer;
  }

  @Override public String toString () {
    return "*(" + pointer + ")";
  }

  @Override public void accept (Visitor v) {
    v.visit(this);
  }
}

File: libs/cppgdsl/src/assembly/operand/memory.cpp
#include "cppgdsl/assembly/operand/memory.h"
#include "cppgdsl/assembly/operand/operand_visitor.h"

namespace gdsl {
namespace assembly {

void memory::put(std::ostream& out) const { out << "*{" << *operand_ << "}"; }

void memory::accept(operand_visitor& ov) const { ov.visit(this); }

bool memory::operator==(operand const& o) const {
  bool equals = false;
  operand_visitor v(true);
  v._([&](memory const* o) { equals = *this->operand_ == *o->operand_; });
  o.accept(v);
  return equals;
}

}  // namespace assembly
}  // namespace gdsl
